### DISCUSSION TITLE
Special characters

### DISCUSSION THREAD

**Discussion by ghost** [matamorphosis/Scrummage#40] [Q&A] [answered]
Are special characters going to be allowed? as they are very useful in getting better results in say Google dorks

**Comment by matamorphosis** [comment, marked as answer]
I thought I had whitelisted all necessary google dorking characters but I was mistaken, the system in place has an explicit deny of several special characters for the security of the platform, where specific special characters are permitted on a case by case basis for each plugin. I have gone ahead and whitelisted Google Dorking special characters, you can always check if there are whitelisted characters by opening up the `plugin_definitions.py` file in the `app` directory, finding the relevant plugin name and check for the presence of a Keyword Argument called `Safe_Characters`, refer to the below screenshot:
  
![Safe_Chars](https://user-images.githubusercontent.com/26025152/142518415-c6d0c353-0963-48f6-b50b-c2404b40b933.png)

You are naturally welcome to make any changes locally as you need to this file by either expanding an existing whitelist for a plugin, or adding the `Safe_Characters` keyword argument to a new plugin, just ensure you add it in a list format, and the keyword is case-sensitive. Failing to add it properly will most likely cause issues when creating or running a task relevant to the plugin you've altered.

You are also welcome to recommend whitelists to the master repo; however, this decision will lie with me, there are some special characters I will never permit to ensure the platform is not susceptible to client-side attack. Such as `<`, `>`, and `%`.
